parsl/executors/flux/executor.py

```diff
@@ -0,0 +1,203 @@
+        Working dir to be used by the executor.
+
+    label : str
+        Label for this executor instance.
+
+    flux_handle_args: collections.abc.Sequence
+        Positional arguments to `flux.Flux()` instance, if any
+
+    flux_handle_kwargs: collections.abc.Mapping
+        Keyword arguments to pass to `flux.Flux()` instance, if any.
+        For instance, the `url` argument to `flux.Flux` can be used to connect to a
+        remote Flux instance at the given URL.
+    """
+
+    def __init__(
+        self,
+        managed: bool = True,
+        working_dir: Optional[str] = "flux_workdir",
+        label: str = "FluxExecutor",
+        flux_handle_args: Sequence = (),
+        flux_handle_kwargs: Mapping = {},
+    ):
+        super().__init__()
+        self.label = label
+        self.working_dir = os.path.abspath(working_dir) if working_dir else os.getcwd()
+        self.managed = managed
+        self._flux_executor = None
+        self.flux_handle_args = flux_handle_args
+        self.flux_handle_kwargs = flux_handle_kwargs
+        self._provider = LocalProvider()
+
+    def start(self):
+        """Called when DFK starts the executor when the config is loaded."""
+        if not _FLUX_AVAIL:
+            raise OptionalModuleMissing(
+                "flux", "Cannot initialize FluxExecutor without flux module"
+            )
+        self._flux_executor = flux.job.FluxExecutor(
+            handle_args=self.flux_handle_args, handle_kwargs=self.flux_handle_kwargs
+        )
+        os.makedirs(self.working_dir, exist_ok=True)
+
+    def shutdown(self):
+        """Shut down the executor, preventing further calls to `submit`."""
+        self._flux_executor.shutdown()
+
+    def submit(self, func, resource_spec, *args, **kwargs):
+        """Wrap a Parsl task in a Flux job and submit it to Flux."""
+        tid = str(uuid.uuid4())
+        infile = os.path.join(self.working_dir, tid + ".pkl")
+        outfile = os.path.join(self.working_dir, tid + ".out.pkl")
+        try:
+            fn_buf = pack_apply_message(
+                func, args, kwargs, buffer_threshold=1024 * 1024
+            )
+        except TypeError:
+            raise SerializationError(func.__name__)
+        with open(infile, "wb") as infile_handle:
+            infile_handle.write(fn_buf)
+        jobspec = flux.job.JobspecV1.from_command(
+            command=[sys.executable, _worker_path, "-i", infile, "-o", outfile],
+            num_tasks=resource_spec.get("num_tasks", 1),
+            num_nodes=resource_spec.get("num_nodes"),
+            cores_per_task=resource_spec.get("cores_per_task", 1),
+            gpus_per_task=resource_spec.get("gpus_per_task"),
+        )
+        jobspec.cwd = os.getcwd()  # should be self.working_dir?
+        jobspec.environment = dict(os.environ)
+        jobspec.stdout = os.path.abspath(
+            os.path.join(self.working_dir, tid + "_stdout_stderr.txt")
+        )
+        jobspec.stderr = jobspec.stdout
```

Comment:
Should stdout and stderr be separated?

Reply:
I feel like I care about this because I've always had it these separate, but I do not know whether a user would care. Does flux default to merging the streams?

Reply:
No, Flux defaults to storing it in its KVS. I'm happy to make them separate, although I'm used to have them united on the terminal, which is why I united them here. I think we should do whatever Parsl does elsewhere, if it does anything similar.

Reply:
We agreed in a conversation to separate them.